flixel/util/FlxCollision.hx

```diff
@@ -177,6 +177,132 @@ class FlxCollision
 
 		return hit;
 	}
+	
+	/**
+	 * A Collision check between two FlxSprites: the first sprite it will take into account the pixels of the sprite, using rotations and scale, etc, 
+	 * but will ONLY look at the Hitbox (width/height + offset) of the second sprite. It will do a bounds check first, and if that passes it will run a
+	 * pixel perfect match on the intersecting area. May be slow, so use it sparingly.
+	 *
+	 * @param	Contact			The first FlxSprite to test against - this sprite will use its pixels
+	 * @param	Target			The second FlxSprite to test again, sprite order is IMPORTANT: this sprite will ONLY use its hitbox (width/height + offset)
+	 * @param	AlphaTolerance	The tolerance value above which alpha pixels are included. Default to 1 (anything that is not fully invisible).
+	 * @param	Camera			If the collision is taking place in a camera other than FlxG.camera (the default/current) then pass it here
+	 * @return	Whether the sprites collide
+	 */
+	@:access(flixel.animation.FlxAnimationController._prerotated)
+	public static function pixelPerfectHitboxCheck(Contact:FlxSprite, Target:FlxSprite, AlphaTolerance:Int = 1, ?Camera:FlxCamera):Bool
+	{
+		// if either of the angles are non-zero, consider the angles of the first sprite in the check
+		var advanced = (Contact.angle != 0 && Contact.animation._prerotated == null) || Contact.scale.x != 1 || Contact.scale.y != 1;
+
+		Contact.getScreenBounds(boundsA, Camera);
+		Target.getScreenBounds(boundsB, Camera);
```

Comment:
it's important to distinguish established flixel terms here:
Hitbox - The collision bounds of a FlxObject, defined by x, y, width and height, not it's angle, scale or anything else
Screen bounds - the graphical space of a FlxSprite, determined by it's position, scale, scrollFactor, offset, origin, angle and any other property that affects the rendering of a FlxSprite (width, height do not directly affect this)

This function seems to be a pixelPerfectCheck that ignores the alpha channel of the second graphic, rather than a comparison between a graphic and a hitbox

Reply:
also is there a test file that shows this in action? where I can move and rotate 2 sprites to visually verify that it does what it says it does in all cases (that is once it correctly explains what it does using established terms)